pandas/tseries/index.py

```diff
@@ -1581,10 +1590,14 @@ def insert(self, loc, item):
         """
 
         freq = None
+
+        if isinstance(item, np.datetime64):
+            # np.datetime64 doesn't have tz, once convert to Timestamp
+            item = Timestamp(item)
```

Comment:
yeh 1 is better, this might work

``vzone = tslib.get_timezone(getattr(Timestamp(other), 'tzinfo', '__no_tz__'))``